update website publishing info (#25)
317brian/druid: distribution/asf-release-process-guide.md

### Update druid.staged.apache.org

This repo (`druid`) is the source of truth for the Markdown files. The Markdown files get copied to `druid-website-src` and built there as part of the release process. It's all handled by a script in that repo  called `do_all_things`.

For more thorough instructions and a description of what the `do_all_things` script does, see the [`druid-website-src` README](https://github.com/apache/druid-website-src)

1. Pull https://github.com/apache/druid-website and https://github.com/apache/druid-website-src. These repositories should be in the same directory as your Druid repository that should have the release tag checked out.

2. From `druid-website`, checkout a staging branch based off of the `asf-staging` branch.

3. From `druid-website-src`, create a release branch from `master`, such as `27.0.0-staging`.

>    1. Update the version list in `static/js/version`.js with the version you're releasing and the release date. The highest release version goes in position 0. 

Reviewer: ```suggestion
   1. Update the version list in `static/js/version.js` with the version you're releasing and the release date. The highest release version goes in position 0. 
```

>     All these files should be part of your PR to `druid-website-src`.
>    <br />

Reviewer: ```suggestion
    All these files should be part of your PR to `druid-website-src`.  
```

1. Make sure you have the latest commits from the `druid` release tag. There have likely been backported documentation PRs between the initial staging date and the release date.

2. Pull `https://github.com/apache/druid-website` and `https://github.com/apache/druid-website-src`. These repositories should be in the same directory as your Druid repository that should have the release tag checked out.

3. From `druid-website`, create a release branch based on the latest `asf-site` branch.

4. From `druid-website-src`, check out a new branch for the release:
   
   1. Update the version list in `static/js/version.js` with the version you're releasing and the release date. The highest release version goes in position 0. Make sure to remove older releases. We only keep the 3 most recent listed in the file.
   1. In `scripts`, run: 
   
   ```python
   # This copies the Markdown files from `druid` and builds the website using them
   # Include `--skip-install` if you already have Docusaurus 2 installed in druid-website-src. 
   # The script assumes you use `npm`. If you use `yarn`, include `--yarn`.

   python do_all_things.py -v VERSION --source /my/path/to/apache/druid
   ```
   

> 5. Add the newly built files to the PR to the src repo (https://github.com/apache/druid-website-src) for the release branch. In the changed files, you should see the following:

Author: ```suggestion
5. Add the files to a PR to the src repo (https://github.com/apache/druid-website-src) for the release branch you just created. In the changed files, you should see the following:
```

  - In `published_versions` directory: HTML files for `docs/VERSION` , `docs/latest`, and assorted HTML and non-HTML files. 
  - In the `docs` directory at the root of the repo, the new Markdown files.
    

>     All these files should be part of your PR to `druid-website-src`.
>    <br />

Reviewer: ```suggestion
    All these files should be part of your PR to `druid-website-src`.  
```
Markdown isn't being rendered properly in the "Verify the site..." line

![image](https://github.com/317brian/druid/assets/7747997/888a347d-2b8b-4e9b-87b2-435392941e1a)
